parsons/databases/table.py

```diff
@@ -164,3 +165,70 @@ def truncate(self):
 
         self.db.query(f"TRUNCATE TABLE {self.table}")
         logger.info(f"{self.table} truncated.")
+
+    def dedup_table(
+        self,
+        order_by_column_name=None,
+        order_by_direction=None,
+        cascade=False,
+        columns_to_ignore=None,
+    ):
+        """
+        Description:
+            This function re-creates a deduped version of a table by grabbing
+            all columns and inserting those into a partition statement for
+            row_number().
+        Args:
+            order_by_column_name: str
+                Column name of specific column that you would like to dedup using order by
+            order_by_direction: str
+                Order by direction, if you would like to dedup by ordering by a specific column,
+                this is the direction of the order by
+                example: 'asc'
+            cascade: bool
+                Set to True if you want any dependent views to be dropped -
+                queries will fail if there are dependent views and this is set to False.
+           columns_to_ignore: list
+                List any columns that should be ignored in the dedup
+        """
+        current_timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        run_cascade = "CASCADE" if cascade else ""
+        order_by_column_name = (
+            "random()" if order_by_column_name is None else order_by_column_name
+        )
+        if order_by_direction is None and order_by_column_name is not None:
+            raise Exception("order_by_direction argument is blank")
+
+        columns_list = self.columns
+
+        # remove order_by columns
+        columns_list.remove(
+            order_by_column_name
+        ) if order_by_column_name is not None else None
+
+        # remove ignore columns
+        if columns_to_ignore is not None:
+            for column in columns_to_ignore:
+                columns_list.remove(column)
+
+        partition = ", ".join(columns_list)
+
+        dedup_query = f"""
```

Comment:
In addition to having issues with views, this is going to reset permissions of the table anytime you dedup it, which I think is really counterintuitive. 

What about instead of renaming the table, we do the select query with the filtered row_number calculation into a temp table, wipe the original table, and then copy the data from the temp table back into the original?

Reply:
@Jason94 does this approach fix the views issues too? (I don't know what the views issues are, as a non-expert SQL person, just want to make sure we're not merging buggy code.)